akka-cluster-tools/src/multi-jvm/scala/akka/cluster/client/ClusterClientSpec.scala

```diff
@@ -63,6 +62,115 @@ object ClusterClientSpec extends MultiNodeConfig {
     }
   }
 
+  //#clientEventsListener
+  class ClientListener(targetClient: ActorRef) extends Actor {
+    override def preStart(): Unit = {
+      context.system.eventStream.subscribe(self, classOf[ContactPointChange])
+      targetClient ! GetContactPoints
+    }
+
+    def receive: Receive =
+      starting(Set.empty[ActorSelection])
+
+    def starting(contactPoints: Set[ActorSelection]): Receive = {
+      case ContactPointAdded(cp, `targetClient`) ⇒
+        context.become(starting(contactPoints + cp))
+      case ContactPointRemoved(cp, `targetClient`) ⇒
+        context.become(starting(contactPoints - cp))
+      case ContactPoints(cps) ⇒
+        context.become(started(contactPoints ++ cps))
```

Comment:
Ah no! That's the point! Between the time that you request `GetContactPoints` and get a `ContactPoints` reply, you may also have received other events...